sycl/source/detail/queue_impl.hpp

```diff
@@ -217,20 +217,19 @@ class queue_impl {
   /// queue on construction. If no async_handler was provided then
   /// asynchronous exceptions will be lost.
   void throw_asynchronous() {
-    std::unique_lock<mutex_class> lock(MMutex);
-
-    if (MAsyncHandler && MExceptions.size()) {
-      exception_list Exceptions;
-
-      std::swap(MExceptions, Exceptions);
-
-      // Unlock the mutex before calling user-provided handler to avoid
-      // potential deadlock if the same queue is somehow referenced in the
-      // handler.
-      lock.unlock();
+    if (!MAsyncHandler)
```

Comment:
```suggestion
    if (!MAsyncHandler || !MExceptions.size())
```

Reply:
Nice catch. Though, I believe, you can't access `MExceptions` without proper safety measures i.e. locking the `MMutex`.

Reply:
I am agree